app/service/koledar_google.py

```diff
@@ -0,0 +1,84 @@
+import json
+import os
+import re
+from datetime import date, datetime, timedelta
+
+import requests
+
+from core.domain.dogodek import Dogodek, TipDogodka
+from core.domain.uporabnik import Uporabnik
+from core.service.koledar_service import KoledarService
+
+
+class KoledarGoogle(KoledarService):
+
+	def __init__(self, google_api_key: str):
+		self.google_api_key = google_api_key
+
+	def dogodki(self, zacetek: date, konec: date) -> list[Dogodek]:
+		api_key = os.environ.get('GOOGLE_API_KEY')
+		email_naslov = os.environ.get('EMAIL')
+		zacetek_ti = datetime.combine(zacetek, datetime.min.time()).isoformat() + 'Z'
+		konec_ti = (datetime.combine(konec, datetime.max.time()) - timedelta(microseconds=1)).isoformat() + 'Z'
+		url = f'https://www.googleapis.com/calendar/v3/calendars/{email_naslov}/events?key={api_key}&timeMin={zacetek_ti}&timeMax={konec_ti}'
+		dogodki = []
+		try:
+			response = requests.get(url)
+			response.raise_for_status()  # Error če je request.status kode čez 300
+			data = json.loads(response.text)
+			for event in data['items']:
+				if 'description' in event:
+					string = event['description']
+					parsed_data = self._parsaj_string(string=string)
+					uporabnik = Uporabnik(
+						ime=parsed_data['name'],
+						priimek=parsed_data['name'],
+						email=parsed_data['email'],
+						email_racuna=parsed_data['email_info'],
+						telefon=parsed_data['phone']
+					)
+					dogodek = Dogodek(
+						ime=None,
+						cena=10.0,
+						tip=TipDogodka.INSTRUKCIJE,
+						zacetek=datetime.fromisoformat(event['start']['dateTime']),
+						konec=datetime.fromisoformat(event['end']['dateTime']),
+						uporabnik=uporabnik,
+						placnik=parsed_data['email_info'],
+						placano=False
+					)
+					dogodki.append(dogodek)
+				else:
+					print('no description, continue to next google event')
+					continue
+		except requests.exceptions.HTTPError as e:
+			print(f"An HTTP error occurred: {e}")
```

Comment:
Namesto printov lahko uporabljaš logging.

Reply:
Sem uporabil logggin kot :
import logging
logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s'),
direkt v classu, ampak za to moram verjetn naredit CONFIG in tam vse nastavitve shranit za  loggerja ?
Iščem projekt k sva ta logger uporabljala, pa kr ne najdem 🙄

Reply:
Aha tukaj se nahaja...
https://github.com/Programerski-klub-Ljubljana/API_programerski-klub.si-deprecated-/blob/438cf564c6fed597c88aa61c2c3c5a55782db8aa/app/APP.py#L4

Reply:
Če dodam v .env LOG_LEVEL='TRACED' dobim :
`ValueError: Unknown level: 'TRACED'` , 
Tako da sem potem dodal kot level kar  "INFO"  in potem deluje 🤔 @urosjarc

Reply:
@ethernal12 TRACED je del knjižnice autologging jo moraš najprej inštalirati potem pa bo šlo.
> https://pythonhosted.org/Autologging/examples-logged.html

Reply:
Sem popravil PR. Upam, da vse, pravilno 🙄🙂.  @urosjarc